lib/binding_web/script/build-sourcemap.ts

```diff
@@ -0,0 +1,62 @@
+import { readFileSync, writeFileSync } from 'fs';
+import { SourceMapGenerator, SourceMapConsumer, RawSourceMap } from 'source-map';
+
+async function fixSourceMap() {
+  const distMap = JSON.parse(readFileSync('dist/tree-sitter.js.map', 'utf8')) as RawSourceMap;
+  const distJs = readFileSync('dist/tree-sitter.js', 'utf8').split('\n');
+  const finalJs = readFileSync('tree-sitter.js', 'utf8').split('\n');
+
+  const lineMap = new Map<number, number>();
+
+  for (let distLine = 0; distLine < distJs.length; distLine++) {
+    const line = distJs[distLine].trim();
+    if (!line) continue;
+
+    for (let finalLine = 0; finalLine < finalJs.length; finalLine++) {
+      if (finalJs[finalLine].trim() === line) {
```

Comment:
This seems reliant on every line of source code being unique, but I don't think that'll be the case. Could we do better by enforcing that `lineMap.get(row)` is monotonically-increasing function of `row`? In other words, rather than taking the *first* matching line, take the first matching line that is *after* any previously-matched line?

Reply:
Oh wow, that is a really good catch - fixing that resulted in the sourcemap being totally accurate now, and coverage data is a lot more richer than before...